client/src/components/Markdown/Markdown.vue

```diff
@@ -30,170 +91,25 @@ const md = MarkdownIt();
-    },
-    history_dataset_type: (actions, args, contents) => {
-        const history_dataset_id = args.history_dataset_id;
-        return `<div class='dataset-type' history_dataset_id="${history_dataset_id}"><pre><code></code></pre></div>`;
-    },
-    tool_stdout: (action, args, content) => {
-        const jobId = args.job_id;
-        return `<div class="tool-stdout" job_id="${jobId}"><pre><code></code></pre></div>`;
-    },
-    tool_stderr: (action, args, content) => {
-        const jobId = args.job_id;
-        return `<div class="tool-stderr" job_id="${jobId}"><pre><code></code></pre></div>`;
-    },
-    job_metrics: (action, args, content) => {
-        const jobId = args.job_id;
-        return `<div class="job-metrics" job_id="${jobId}"></div>`;
-    },
-    job_parameters: (action, args, content) => {
-        const jobId = args.job_id;
-        const param = args.param;
-        return `<div class="job-parameters" job_id="${jobId}" param="${param}"></div>`;
-    },
-    generate_galaxy_version: (actions, args, content) => {
-        return `<div class="galaxy-version"><pre><code></code></pre></div>`;
-    },
-    generate_time: (actions, args, content) => {
-        return `<div class="generate-time"><pre><code></code></pre></div>`;
-    },
-    invocation_time: (actions, args, content) => {
-        const invocationId = args.invocation_id;
-        return `<div class="invocation-time" invocation_id="${invocationId}"><pre><code></code></pre></div>`;
-    },
-};
-
-md.renderer.rules.fence = function (tokens, idx, options, env, slf) {
-    const token = tokens[idx];
-    const info = token.info ? token.info.trim() : "";
-    const content = token.content;
-
-    if (info == "galaxy") {
-        const galaxy_function = FUNCTION_CALL_LINE_TEMPLATE.exec(content);
-
-        const args = {};
-        const function_name = galaxy_function[1];
-        // we need [... ] to return empty string, if regex doesn't match
-        const function_arguments = [...content.matchAll(new RegExp(FUNCTION_CALL, "g"))];
-
-        for (let i = 0; i < function_arguments.length; i++) {
-            if (function_arguments[i] === undefined) continue;
-            const arguments_str = function_arguments[i].toString().replace(/,/g, "").trim();
-
-            if (arguments_str) {
-                const [key, val] = arguments_str.split("=");
-                args[key.trim()] = val.replace(/['"]+/g, "").trim();
-            }
-        }
-
-        return RENDER_FUNCTIONS[function_name](function_name, args, content);
-    } else {
-        return default_fence(tokens, idx, options, env, slf);
-    }
-};
-
-function render_fenced_output(tag, objects, idAttr, metadataKey) {
-    $("." + tag).each((i, el) => {
-        const objectId = $(el).attr(idAttr);
-        const meta = objects[objectId][metadataKey];
-        $(el).find("code").text(meta);
-    });
-}
-
 export default {
+    store: store,
+    components: {
```

Comment:
wow, this is super cool! way way cleaner than calling mount for each component

Reply:
Cool (: Can you take a look at https://github.com/galaxyproject/galaxy/blob/e64712e1fbb840f6b22d1283895e6189e0b26027/client/src/components/Markdown/Elements/HistoryDatasetAsImage.vue. If necessary, I should be able to also use: https://github.com/galaxyproject/galaxy/blob/e64712e1fbb840f6b22d1283895e6189e0b26027/client/src/components/Dataset/DatasetAsImage/DatasetAsImage.vue instead of duplicating the functionality.

Reply:
> Cool (: Thanks for looking through this. I noticed that https://github.com/galaxyproject/galaxy/blob/e64712e1fbb840f6b22d1283895e6189e0b26027/client/src/components/Dataset/DatasetAsImage/DatasetAsImage.vue uses the store while the wrapper I added just uses the `link` see: https://github.com/galaxyproject/galaxy/blob/e64712e1fbb840f6b22d1283895e6189e0b26027/client/src/components/Markdown/Elements/HistoryDatasetAsImage.vue. If necessary, we can adjust or consolidate those two back into one.

Consolidation would be a good idea. Otherwise, we should replace `<img>`  with `<b-img>` in https://github.com/galaxyproject/galaxy/blob/e64712e1fbb840f6b22d1283895e6189e0b26027/client/src/components/Dataset/DatasetAsImage/DatasetAsImage.vue